packages/cli/src/commands/export-dynamic-plugin/backend.ts

```diff
@@ -0,0 +1,320 @@
+
+import path, { basename } from 'path';
+
+import { Output } from '../../lib/builder';
+import { makeRollupConfigs } from '../../lib/builder/config';
+import { embedModules } from '../../lib/builder/embedPlugin';
+import { buildPackage, formatErrorMessage } from '../../lib/builder/packager';
+import { loadCliConfig } from '../../lib/config';
+import { productionPack } from '../../lib/packager/productionPack';
+import { paths } from '../../lib/paths';
+import { Task } from '../../lib/tasks';
+
+export async function backend(
+  roleInfo: PackageRoleInfo,
+  opts: OptionValues,
+): Promise<void> {
+  if (!fs.existsSync(paths.resolveTarget('src', 'dynamic'))) {
+    throw new Error(
+      `Package doesn't seem to support dynamic loading. It should have a ./src/dynamic folder, containing the dynamic loading entrypoints.`,
+    );
+  }
+
+  const outputs = new Set<Output>();
+
+  if (roleInfo.output.includes('cjs')) {
+    outputs.add(Output.cjs);
+  }
+  if (roleInfo.output.includes('esm')) {
+    outputs.add(Output.esm);
+  }
+
+  const pkgContent = await fs.readFile(
+    paths.resolveTarget('package.json'),
+    'utf8',
+  );
+  const pkg = JSON.parse(pkgContent) as BackstagePackageJson;
+  if (pkg.bundled) {
+    throw new Error(
+      'Packages exported as dynamic backend plugins should not have the "bundled" field set to true',
+    );
+  }
+
+  const target = path.join(paths.targetDir, 'dist-dynamic');
+
+  if (
+    !pkg.files?.includes('dist-dynamic/*.*') ||
+    !pkg.files?.includes('dist-dynamic/dist/**') ||
+    !pkg.files?.includes('dist-dynamic/alpha/*')
+  ) {
+    throw new Error(
+      `Package doesn't seem to support dynamic loading: its "files" property should include the following entries: ["dist-dynamic/*.*", "dist-dynamic/dist/**", "dist-dynamic/alpha/*"].`,
+    );
+  }
+
+  const mergeWithOutput: (string | RegExp)[] = [];
+
+  const commonPackage = pkg.name.replace(/-backend$/, '-common');
+  if (commonPackage !== pkg.name) {
+    mergeWithOutput.push(commonPackage);
+  }
+
+  if (opts.embedPackage !== undefined) {
+    for (const pkgToEmbed of opts.embedPackage as string[]) {
+      if (pkg.dependencies === undefined || !(pkgToEmbed in pkg.dependencies)) {
+        console.log(
+          `Embeded package '${pkgToEmbed}' is not part of direct dependencies.Are you sure you want to embed it ?`,
+        );
+      }
+      mergeWithOutput.push(pkgToEmbed);
+      const relatedCommonPackage = pkgToEmbed.replace(/-backend$/, '-common');
+      if (relatedCommonPackage !== pkgToEmbed) {
+        mergeWithOutput.push(relatedCommonPackage);
+      }
+    }
+  }
+
+  const filter = {
+    include: mergeWithOutput,
+    exclude: mergeWithOutput.length !== 0 ? undefined : /.*/,
+  };
+  const moveToPeerDependencies: (string | RegExp)[] = [/@backstage\//];
+
+  if (opts.sharedPackage !== undefined) {
+    for (const sharedPkg of opts.sharedPackage as string[]) {
+      moveToPeerDependencies.push(sharedPkg);
+    }
+  }
```

Comment:
Can't this be simply:

```suggestion
  const moveToPeerDependencies: (string | RegExp)[] = [/@backstage\//, ...(opts.sharedPackage || [])]
```

Reply:
Fixed